modules/cfe_testcase/src/es_info_test.c

```diff
@@ -0,0 +1,263 @@
+    UtAssert_INT32_EQ(CFE_ES_GetAppInfo(&AppInfo, AppId), CFE_SUCCESS);
+
+    UtAssert_INT32_EQ(CFE_ES_GetTaskInfo(&TaskInfo, AppInfo.MainTaskId), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ES_GetTaskID(&TaskId), CFE_SUCCESS);
+    UtAssert_ResourceID_EQ(TaskId, AppInfo.MainTaskId);
+
+    UtAssert_StrCmp(TaskInfo.AppName, AppInfo.Name, "TaskInfo.AppName (%s) = AppInfo.name (%s)", TaskInfo.AppName,
+                    AppInfo.Name);
+    UtAssert_StrCmp(TaskInfo.TaskName, AppInfo.MainTaskName, "TaskInfo.TaskName (%s) = AppInfo.MainTaskName (%s)",
+                    TaskInfo.TaskName, AppInfo.MainTaskName);
+
+    UtAssert_ResourceID_EQ(TaskInfo.TaskId, AppInfo.MainTaskId);
+    UtAssert_ResourceID_EQ(TaskInfo.AppId, AppId);
+    UtAssert_INT32_EQ(TaskInfo.ExecutionCounter, AppInfo.ExecutionCounter);
+
+    UtAssert_INT32_EQ(CFE_ES_GetTaskInfo(&TaskInfo, CFE_ES_TASKID_UNDEFINED), CFE_ES_ERR_RESOURCEID_NOT_VALID);
+    UtAssert_INT32_EQ(CFE_ES_GetTaskInfo(NULL, CFE_ES_TASKID_UNDEFINED), CFE_ES_BAD_ARGUMENT);
+    UtAssert_INT32_EQ(CFE_ES_GetTaskID(NULL), CFE_ES_BAD_ARGUMENT);
+}
+
+void TestLibInfo(void)
+{
+    CFE_ES_LibId_t   LibIdByName;
+    CFE_ES_AppInfo_t LibInfo;
+    const char *     LibName     = "ASSERT_LIB";
+    const char *     InvalidName = "INVALID_NAME";
+    char             LibNameBuf[OS_MAX_API_NAME + 4];
+
+    UtPrintf("Testing: CFE_ES_GetLibIDByName, CFE_ES_GetLibName, CFE_ES_GetLibInfo");
+
+    UtAssert_INT32_EQ(CFE_ES_GetLibIDByName(&LibIdByName, LibName), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ES_GetLibInfo(&LibInfo, LibIdByName), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ES_GetLibName(LibNameBuf, LibIdByName, sizeof(LibNameBuf)), CFE_SUCCESS);
+    UtAssert_StrCmp(LibNameBuf, LibName, "CFE_ES_GetLibName() = %s", LibNameBuf);
+    UtAssert_True(LibInfo.Type == CFE_ES_AppType_LIBRARY, "Lib Info -> Type = %d", (int)LibInfo.Type);
+    UtAssert_StrCmp(LibInfo.Name, LibName, "Lib Info -> Name = %s", LibInfo.Name);
+    UtAssert_StrCmp(LibInfo.EntryPoint, "CFE_Assert_LibInit", "Lib Info -> EntryPt  = %s", LibInfo.EntryPoint);
+    UtAssert_StrCmp(LibInfo.FileName, "/cf/cfe_assert.so", "Lib Info -> FileName = %s", LibInfo.FileName);
+    UtAssert_True(LibInfo.StackSize == 0, "Lib Info -> StackSz  = %d", (int)LibInfo.StackSize);
+
+    if (LibInfo.AddressesAreValid)
+    {
+        UtAssert_True(LibInfo.AddressesAreValid > 0, "Lib Info -> AddrsValid?  = %d", (int)LibInfo.AddressesAreValid);
+        UtAssert_True(LibInfo.CodeAddress > 0, "Lib Info -> CodeAddress  = %d", (int)LibInfo.CodeAddress);
+        UtAssert_True(LibInfo.CodeSize > 0, "Lib Info -> CodeSize  = %d", (int)LibInfo.CodeSize);
+        UtAssert_True(LibInfo.DataAddress > 0, "Lib Info -> DataAddress  = %d", (int)LibInfo.DataAddress);
+        UtAssert_True(LibInfo.DataSize > 0, "Lib Info -> DataSize  = %d", (int)LibInfo.DataSize);
+        UtAssert_True(LibInfo.BSSAddress > 0, "Lib Info -> BSSAddress  = %d", (int)LibInfo.BSSAddress);
```

Comment:
Same here - addresses should be unsigned long.

Reply:
fixed